### ISSUE TITLE
FUTBIN banned due to FUT Enhancer?

### ISSUE THREAD

**Issue by Lurd67** [Mardaneus86/futwebapp-tampermonkey#167]
Hi everyone, 

I have been using this extension for months without any issue (thank you Mardaneus86, you helped so many of us), and I was pleasantly surprised to see that FUTBIN prices would now be displayed on SBC screens.

However, about an hour later, I realized that I could not access FUTBIN anymore, being greeted by an Error 403. I contacted FUTBIN support to know what was wrong, and they told me I was IP banned for some reason. They asked if I was using any tools or extensions that sent the website requests, and I replied that yes, I was using FUT Enhancer (no point in lying).

I was then immediately unbanned, but I was also warned that the system could ban me again, should I keep using the extension.

Is anyone else in this situation? This script is amazing, and it saved me from countless headaches, but if it gets users banned on FUTBIN (which is the only reliable source for PC prices, I can't do anything without it), I will have to drop it.

Thanks!

**Comment by Mardaneus86** [comment]
I just double checked and the player prices are fetched after each player position change on the SBC page. Would be surprised if it is related to the SBC page.
Maybe it is still related to the bug that was introduced after a Futbin change where more than 30 players on a page caused repeated calls to Futbin? This has been fixed since v1.4.2 and upwards.

**Comment by Lurd67** [comment]
Hmm I don't know, I wasn't aware of this previous issue.
I don't know anything about development, so I can't really help you troubleshoot. Might be linked to the SBC page, might not be. I just thought you'd like to be aware of such an issue :)

**Comment by Mardaneus86** [comment]
Added a setting where you have to enable the SBC and Squad prices for Futbin explicitly.

For how long have you been using the web app today/yesterday? Might be that you reached some kind of request limit on Futbin that I'm not aware of.

**Comment by nokai** [comment]
I commented in this case
https://github.com/Mardaneus86/futwebapp-tampermonkey/issues/166

It's not because of the sbc page. I only use futbin in the tradepile.
Many requests ban you the ip.

**Comment by Mardaneus86** [comment]
I am aware you added, but without saying which version you are using I can't determine the cause of it 😒 So I was assuming it was related to using v1.4.1 and earlier.

Anyway, just released an update v1.5.1 that adds a setting for using Futbin prices on SBC and Squads screens explicitly.

**Comment by Lurd67** [comment]
> Added a setting where you have to enable the SBC and Squad prices for Futbin explicitly.
> 
> For how long have you been using the web app today/yesterday? Might be that you reached some kind of request limit on Futbin that I'm not aware of.

Great, that option makes me less worried, for the moment.
Hmm I haven't used it that much. I spent most of the evening (CET time zone, Prague) watching the Europa League until Slavia beat Sevilla (so 12-ish), and I only did about 4 or 5 SBCs this morning, while I have has days in the past where I did hundreds.

Here is the reply I got to my initial email:

> Hello,
> 
> It means that for some reason our system banned your IP. Please send us your public IP address and we will check why. 
> 
> Have you used a tool or an extension that sends requests to our site?  If so, we need you to uninstall it before we can lift the ban.

Let me know if you need any more info, I think the SBC feature is amazing.

**Comment by Mardaneus86** [comment]
If anyone has more information, please discuss in this thread.

Could be that they have a delayed ban system. The v1.4.1 and earlier spammed the servers a bit due to a change on Futbins side, unfortunately.

**Comment by Lurd67** [comment]
Another user contacted me on Reddit to let me know he has a similar issue. I sent him the link to this thread so he could participate. Hopefully he'll be here shortly

**Comment by castboom** [comment]
Noticed futbin being blocked on my IP at around 4:26pm central time yesterday. Have never had any issues before. I was most likely on version 1.5 at the time of the issue. After reading this thread and sending an email to futbin letting them know I was using this tool they responded with “You shouldn’t use it, it’s not safe”. They didn’t unblock my IP either unfortunately.

**Comment by Mardaneus86** [comment]
Thanks for the information. I have a feeling it's not related to the update at all, but they just started enforcing more strict policies since yesterday. Another user contacted me who was using a similar thing, but not this plugin, and he also has the same problem.

Regarding the "it's not safe", that's absolutely not true. The source code is totally open. So if anything would be unsafe, someone else would have mentioned it already.

However, if they are indeed enforcing stricter policies, that might mean the end of this script. I would love to work together with Futbin on this, but they weren't open for that in the past. Probably because every page view earns them money (hence that's why I added the `Show player on Futbin` button in the first place).

**Comment by Lurd67** [comment]
I'd hate to see your script go, it's possibly the single most useful tool I'm using on a day-to-day basis. I will try to limit my use and keep an eye on things to know if there's anything new.
But again, thanks for all your work.

**Comment by castboom** [comment]
they just responded to my email after I asked why it's not safe:

Hi,
We've lifted the ban and you should be able to access our servers now.
Regarding "why It's not safe" - it will get you banned from our servers. 
Best,
The FUTBIN

**Comment by Mardaneus86** [comment]
I know, I love to work on this script, but I don't want people to get bans on Futbin or any other system either. My main focus has been to limit stress on the Futbin side from the start, and linked to their service to make sure they still get their ad income.

But it's not worth my time to fight these kinds of things. I've always been working on this with the best intentions. The above message from Futbin says enough in my opinion.

**Comment by Lurd67** [comment]
Now that they have released Premium plans for mobile, maybe you can contact them again an offer some partnership to include it in their subscriptions for a percentage of the income? I can imagine it would entice a lot of people to subscribe, and it would limit the weight on their servers.
Of course, I assume you want to keep this tool free for everyone (and thank you for that), but it could be mutually beneficial in my opinion. You could keep the script running without the fear of them suddenly hardening their policies, and they would have an amazing tool at their disposal.

**Comment by castboom** [comment]
I bet its against EA's TOS and I know EA would not allow something like this as some users have been banned from EA for using other extensions so I doubt they even want to touch this.

**Comment by Mardaneus86** [comment]
They will never share percentages of income 😄 And indeed I want to keep this tool free. If they would state the policies/limits, I can add that to the script. But they probably just want you to go to their website for each player, getting them a lot of revenue on it.

No one has been banned on EA because they were using this script as far as I know. By the way, their price scraping tools + club importer also do stuff that's against EA's ToS. I doubt they do price scraping by hand...

**Comment by castboom** [comment]
Sorry yeah, I didn't mean this extension but I've seen on twitter some other traders that have been banned and EA's response was for using other browser web tools/extensions.

**Comment by nokai** [comment]
Maybe if you put a button for whoever wants to ask the prices to futbin.

**Comment by castboom** [comment]
Isn’t that already there? “Futbin integration”

**Comment by SpookyNun** [comment]
Happened to me Aswell, today at approximately 1 AM GMT.
I was just searching player prices on the transfer market and it was t popping up with the Futbin bin price, so I check FUTBIN and it wasn’t working, I thought FUTBIN was down, turns out my IP was blocked, I contacted Futbin and they lifted the ban, i don’t know whether to carry on using it, because FUTBIN is the best source of player prices etc.

**Comment by nokai** [comment]
> Happened to me Aswell, today at approximately 1 AM GMT.
> I was just searching player prices on the transfer market and it was t popping up with the Futbin bin price, so I check FUTBIN and it wasn’t working, I thought FUTBIN was down, turns out my IP was blocked, I contacted Futbin and they lifted the ban, i don’t know whether to carry on using it, because FUTBIN is the best source of player prices etc.

If you restart the router. Change IP and you can see FUTBIN.

**Comment by sparcd** [comment]
This happened to me today (running 1.5.1 with only the basic FUTBIN integration enabled). 
 Fortunately, support@futbin.com removed the ban for me quickly, however they did not comment on my offer to pay for a subscription (to support the elevated traffic).

Would it be possible to redesign FUT Enhancer around an intermediate/cached data source such as Google Docs?  Clients would check there first and only refresh from FUTBIN on a manual request or for stale data.  I would think combining FUTBIN data with the **Search minimum BIN** function (which I use all the time!) to populate the datastore would yield a more accurate system, given FUTBIN prices are often hours out of date.

**Comment by Imit4toR** [comment]
Guess it makes sense now to take 'Show link to player page' out of 'Futbin Integration' option so we could switch them separately.

**Comment by sunvenom** [comment]
Hi, much respect for this fine extension, but I have the same issue as well. IP was perma-banned, and after renewing IP was able to access again. FYI, I've never used the SBC or Squad price functions. Only for tradepile & searching.

I was expecting this to kick in at some point, since this extension seems to be designed quite aggressively  (ie. duplicate players aren't optimized in requests, insta-retry loops upon error, data refetch attempt on every single screen reload)

And I was thinking about creating a local cache specifically to address these issues (optimize query / delay retries / cache results and refresh only after set period of time), but if this were to become a community effort as @sparcd recommends, I'm all in.

**Comment by SpookyNun** [comment]
> > Happened to me Aswell, today at approximately 1 AM GMT.
> > I was just searching player prices on the transfer market and it was t popping up with the Futbin bin price, so I check FUTBIN and it wasn’t working, I thought FUTBIN was down, turns out my IP was blocked, I contacted Futbin and they lifted the ban, i don’t know whether to carry on using it, because FUTBIN is the best source of player prices etc.
> 
> If you restart the router. Change IP and you can see FUTBIN.

I have a static IP, so it doesn’t restart once I reset it.

**Comment by Mardaneus86** [comment]
@sparcd I like the idea, but how would we prevent abuse on using a service without authentication? Anyone would be able to inject false prices or potentially remove prices. If you have ideas on that, please contact me via Gitter so we can discuss in detail.

I could setup a custom system, but that would mean I need to bear the costs of it, and have a way of authentication in order to ban users that do false price reports. If we can find a decent solution as a community, I would like to facilitate that very much.

Having an intermediate cache doesn't really help, since prices can update at any point in time. However, having a button to request the prices might be helpful and reduce stress on Futbins side.

@sunvenom duplicate players are filtered in a request, however if the next screen contains the same player it is requested again.

**Comment by Mardaneus86** [comment]
Just released an update (v1.5.2) that takes the updated Futbin url into account and also adds a 5 minute wait period in case a call to Futbin fails in order to prevent request spam. A notification will warn you when that happens.

**Comment by danielanguita** [comment]
FYI - I was using Transfer List and seems that I got banned from Futbin... :(

**Comment by omasseau** [comment]
After updating to 1.5.2, I just got banned today.
I've only use the transfer list.

**Comment by Mardaneus86** [comment]
How much did you use the api? Without extra information I can't do anything about it.

**Comment by metabrain** [comment]
Hey,

While I don't use this extension, I have my own that I'm developing for personal use and I have ran into the exact same issue as you state in this thread.. Plus I am optimising now in order to avoid duplicate requests/hammering their API but the bans are already in place. 5 days so far and still no un-ban. Will I have to send them an email?

Cheers,
Dan

**Comment by danielanguita** [comment]
I did sent them an email and got un-banned.
Seems that futbin is now throttling queried and banning when abuse... Idk
how you guys have this implemented, but basically we need to do one call at
a time slowly and try to cache in the client side to avoid multiple calls.
Let me know if I can help in the development, happy to do so.

**Comment by Mardaneus86** [comment]
@metabrain I am not doing Futbin support, nor support for personal projects even though they are in the same area.

I am thinking about the proposal to add a button to fetch the prices on request instead of automatically. It depends on how heavily you use the web app. Caching brings another issue that I'd need to setup how much the cache can get out of date. Some player prices are updated often, others only every few hours.

**Comment by danielanguita** [comment]
I see your point Tim, I think at the end we need to make sure we avoid
getting banned. The button might be a good option, but would be even better
to have a "smart" way to refresh players prices following futbin
restrictions. I'm asking them if they can provide their limits so we can
take that in count. I use the webapp only because of your plugin :)

**Comment by hooleesheettv** [comment]
biggest issue atm is that u still get banned with the new version :( 
I rly love your tool, hope u guys gind a way. I use the web app a lot because of your tool, but futbin banns now rly fast (2-4 hours after using it.

**Comment by Mardaneus86** [comment]
Please let me know how long you have been using the app before experiencing a Futbin ban. As I said multiple times before, I can't do anything about it without more information.

**Comment by hooleesheettv** [comment]
i used it for 2 hours and after i came back to the pc i was banned (4 hours after i think) I got banned 2 times allready. I deactivated it for now. 
Man u did great work not your fault futbin hast now those strict limits.

**Comment by blackcalpac** [comment]
Have a good day. They sent an reply to my mail:
“Hello,

It means that for some reason our system banned your IP. Please send us your public IP address and we will check why. 

Have you used a tool or an extension that sends requests to our site?  If so, we need you to uninstall it before we can lift the ban.

You can get your IP in here: https://www.whatismyip.com/“

I don’t know what can ı do. I havent sent a reply yet.

**Comment by zoma24** [comment]
really sorry to see what is happening, this tool was very helpful and I loved it.
I just wanted to let you know i was using v1.5.2 and selling some packed players on the tradepile. IP got banned just a few minutes after I started. 
I hope you guys  can find a solution.

**Comment by castboom** [comment]
Has anyone that works on this reached out to FUTBIN to see if they can work something out? Not sure how nice they are but it would be worth a shot. I don't know the programming lingo to ask them the right questions. 

What about using another price site like: https://www.futview.com/analytics/ratings

**Comment by richardx95** [comment]
I came across the same bann from Futbin. The easy way to fix it is to just install a VPN extension to your webbrowser and select another country. That will give you new access to Futbin.

**Comment by richardx95** [comment]
I also contacted Futbin with the error code and they responded with this:

> Hi,
> 
> It spams our servers and if you continue using it our system will auto ban you again.
> We've lifted the ban and you should be able to access our servers now. 
> 
> Best,
> The FUTBIN Team

Hope this gives you all more information.

**Comment by jackyf13** [comment]
Same issue here, though when I get banned from Futbin I've been going to my router settings and disconnecting then reconnecting to renew my IP address and then the ban is lifted again, hope this helps some of you

**Comment by Feigenstaub** [comment]
I just put some players on the market and after like 20 I got constantly the futbin error and futbin banned me aswell.

**Comment by Avalius1987** [comment]
Been using fut enhancer since Oct last year. Updated to 1.5.2 when you released it and now I'm banned from futbin to.
I am getting this message:
Error - 403. Your client does not have permission to access this page at this time. Contact us on support@futbin.com
Still works on my phone

**Comment by ehakdem** [comment]
hello, i also got banned from futbin. i'm sure it's not related about the versions of the script. i guess futbin started to apply an aggressive ban policy for many requests per some limited time. An alternative solution can be saving the query results to a temporary storage and reading from there for a while. Or just need to modify the scripts to do less query.

for the people whom doesnt have static ip can reset their modem or mobile network connection and have continue to fun with this great script :) but for the people whom has static ip should contact with the futbin support or need to use vpn.

somehow i temporarily needed to disable this script cos i also use another script which also use futbin so its also banned :)  so its really suprising that we dont get any ban from ea but ban from futbin locked us :)

**Comment by Mardaneus86** [comment]
The cause for this issue is known, so I'm closing the discussion since it spams my inbox a bit too much to keep focus 😄. If you have useful insights beyond what is mentioned here already, please contact me via Gitter.

I am looking into multiple ways of solving this
- Button to request price on demand for specific player (makes the bargain and totals scripts fairly pointless)
- Button to request prices for a specific page on demand
- Other source for player prices

A few workarounds for when you have been banned have been mentioned in this topic. Feel free to use those until this has been resolved.
